### PR TITLE
Flow reorganization

### PR DESCRIPTION
**Pull request by RackhamLeNoir**

### CODE REVIEW

File: template/chat.html [hunk 1]
```diff
@@ -34,6 +34,9 @@
         }
       },
 
+      /*
+        writes a message in the chat box.
+      */
       sendMessage: function(user, message) {
         if (message != '')
```
Comment (tania-m) [inline template/chat.html]: This is q dangerous check to do. If message == "  " (two whitespaces), the test will pass. So, as it is, it implies at least that the string has been trimmed beforehand. 
Otherwise, we should use underscore.js, underscore.string.js plug-in (https://github.com/epeli/underscore.string) and the isBlank method for strings (that avoids that we have to rewrite our own).

Comment (RackhamLeNoir) [inline template/chat.html]: A regular expression to test if a string is empty. That's ridiculous. OO

Comment (tania-m) [inline template/chat.html]: That also depends on what an emptu string is for you. But, depending on the empty string definition, you might end up adding more and more tests (I am not defending the regex solution, just saying that thinking and checking all edge cases might not be as trivial as it may look like).

Comment (RackhamLeNoir) [inline template/chat.html]: I'm old school. To me an empty string is a non null pointer to '\0'. Why not having to care about spaces if you want to specifically? It doesn't make sense to me. Anyway, we cannot change this behaviour so I guess we have to deal with it.

Comment (tania-m) [inline template/chat.html]: I would check the string is not undefined and not null to start. Then the trickery with the empty spaces, at least trimming at start and at end and checking if is it is different from the empty string. At least.

Comment (nicolasnoble) [inline template/chat.html]: 1. Why would we want to prevent empty messages ? :-D
2. You may rather want something like typeof(message) == "string"
3. The type is going to be converted anyway; if typeof(message) == "number", that'd still work. It all depends on the API contracts we want "sendMessage" to impose.
4. That's not part of the overall changes :P

File: template/clientsettings.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          type: String,
+          notify: true
+        }
+      },
+
+      behaviors: [
+        PapanLocalize
+      ],
+
+      _toggle: function() {
```
Comment (nicolasnoble) [inline template/clientsettings.html]: That doesn't seem to be used anywhere - and that's counter-intuitive with a radio-group mechanism.

File: template/lobby.html [hunk 3]
```diff
@@ -34,7 +34,9 @@
             view.addEventListener('action', (event) => {
               ipc.send('asynchronous-message', { type: 'action', data: event.detail })
             })
-            document.getElementById('main-view').appendChild(view)
+            let mainview = document.getElementById('main-view')
+            while (mainview.hasChildNodes()) mainview.removeChild(mainview.lastChild);
```
Comment (nicolasnoble) [inline template/lobby.html]: Because of the way polymer registers elements, we won't be able to dynamically load more than one game board. If the same game is started again, this codepath will work fine, but if we are switching to a different game board, we need to reload the page in order to purge the currently dynamically loaded template. `location.reload()` is the way to do that.

More specifically, above, we must first remember which game template was previously loaded:

```
let loadedGame
```

Then, after the server sends us the game information:

```
if (game.gamePath == loadedGame) {
  location.href = 'something-to-auto-start-game'
  location.reload()
} else {
  loadedGame = game.gamePath
[...]
```

That's going to be more testable once we have more than one board.

File: template/clientsettings.html (hunk 2 above)
Comment (RackhamLeNoir) [inline template/clientsettings.html]: This is for internationalization. Try removing it, but I doubt it will be working anymore.

File: template/lobby.html (hunk 3 above)
Comment (RackhamLeNoir) [inline template/lobby.html]: I let you handle this.
